### DISCUSSION TITLE
Assign users to groups based on SSO-provided information

### DISCUSSION THREAD

**Discussion by multipleofzero** [outline/outline#6379] [Ideas]
It would be great if it was possible to use information from SSO (OpenId Connect) to assign users to groups in Outline.

Wiki.js for example provides an option called "Map Groups" that allows to check a (configurable) OpenId Connect claim (often times "member_of") for values which correspond to group names the user should be assigned to. Confluence, Youtrack and others use similar approaches to ease the burden of managing users and groups in their applications.

The scenario I have is:
- users are configured in an LDAP directory
- users have groups assigned in LDAP
- SSO passes these groups (can pass them) in a "member_of" claim to services the users access
- users in Outline must still be manually added (or removed) from groups in Outline

The main benefit (for us) would be if users could be automatically added to groups upon creation; being able to later change (add/remove) group assignments via LDAP/SSO claims would be a plus but is less essential.

**Comment by tommoor** [comment]
Just a matter of time, will take a look at how wikijs implemented this feature

**Reply by iamrubeng** [reply]
That's a really needed feature. In our company we need to sync AzureAD groups. Now we use the API to do this.

**Reply by Blendman974** [reply]
Wasn't pull request #6581 the beginning of a solution to this discussion?
This feature is very useful indeed, especially as an external authentication source is required to operate outline it would be ideal to have this external source manage the groups

**Comment by pzasuwa** [comment]
Hi, Is there any update regarding implementing group sync from SSO?
